themes/default/content/blog/kubernetes-fundamentals-part-two/index.md

```diff
@@ -0,0 +1,215 @@
+
+Now just run `pulumi up` to preview the changes, select `yes` when prompted, and watch the cluster stand up! You'll see some work happening in your terminal as well as in your Pulumi account, and when it's all done, you can look at your GCP dashboard and see the cluster there, too.
+
+But we should actually put something in the cluster, shouldn't we?
+
+## A Deployment for the Cluster
+
+Let's work on that Kubernetes provider. Using a Kubernetes Provider allows us to abstract away some of the specifics of operating a cluster on one cloud provider or another. That way, your experience from here on out is fairly consistent regardless of where you're hosting this.
+
+In the same `index.ts` file from before, add this to the end:
+
+```typescript
+// Create a Kubernetes Namespace
+const ns = new k8s.core.v1.Namespace(name, {}, { provider: clusterProvider });
+
+// Export the Namespace name
+export const namespaceName = ns.metadata.apply(m => m.name);
+
+// Create a NGINX Deployment
+const appLabels = { appClass: name };
+const deployment = new k8s.apps.v1.Deployment(name,
+    {
+        metadata: {
+            namespace: namespaceName,
+            labels: appLabels,
+        },
+        spec: {
+            replicas: 1,
+            selector: { matchLabels: appLabels },
+            template: {
+                metadata: {
+                    labels: appLabels,
+                },
+                spec: {
+                    containers: [
+                        {
+                            name: name,
+                            image: "nginx:latest",
+                            ports: [{ name: "http", containerPort: 80 }]
+                        }
+                    ],
+                }
+            }
+        },
+    },
+    {
+        provider: clusterProvider,
+    }
+);
+
+// Export the Deployment name
+export const deploymentName = deployment.metadata.apply(m => m.name);
+
+// Create a LoadBalancer Service for the NGINX Deployment
+const service = new k8s.core.v1.Service(name,
+    {
+        metadata: {
+            labels: appLabels,
+            namespace: namespaceName,
+        },
+        spec: {
+            type: "LoadBalancer",
+            ports: [{ port: 80, targetPort: "http" }],
+            selector: appLabels,
+        },
+    },
+    {
+        provider: clusterProvider,
+    }
+);
+
+// Export the Service name and public LoadBalancer endpoint
+export const serviceName = service.metadata.apply(m => m.name);
+export const servicePublicIP = service.status.apply(s => s.loadBalancer.ingress[0].ip)
+```
+
+Here, we're creating a namespace and adding an NGINX deployment with an accompanying LoadBalancer service and exporting everything we need to run it. Run `pulumi up` once again, double-check the preview, and select `yes` to update the deployment. Now everything's up and running on a managed Kubernetes service that you didn't have to stand up by hand, and won't have to expend as much effort maintaining. 
```

Comment:
Patch to fix linter for preview
```suggestion
Here, we're creating a namespace and adding an NGINX deployment with an accompanying LoadBalancer service and exporting everything we need to run it. Run `pulumi up` once again, double-check the preview, and select `yes` to update the deployment. Now everything's up and running on a managed Kubernetes service that you didn't have to stand up by hand and won't have to expend as much effort maintaining.
```

Reply:
Also, I wonder if it might be helpful to link to the various Kubernetes docs for namespaces, deployments, services, etc. to help tie the abstract to the actual here for folks who aren't familiar with the ops side of things.

Reply:
Good point, yes. I'll add that in a bit.